USSF-ORBIT/ussf-portal-client
# refactor: Refactor MySpace component

# SC-1812



## Proposed changes
- Remove `getMySpace` query from `MySpace` component and instead use `portalUser` from the `authContext` to provide user's myspace data to the component
- Add loading and error checks for the `getUser` query in `DefaultLayout`, `PageLayout` and `ArticleLayout`
- Remove getDisplayName query from `/settings` and use `portalUser` instead


---



## Reviewer notes
How to test:
- Try out as many features in MySpace as you can. The original functionality should still be present, we're just using the `getUser` query instead of `getMySpace`. Add/remove collections/widgets, add/remove bookmarks, add collections from the sites & apps page, update your display name, etc.

### Note (optional reading, not crucial to reviewing):
In our last eng sync I mentioned how I removed all of the `useUser` hooks from each page because it was seemingly unnecessary. I still believe those checks are unnecessary to a degree, but there was some strange behavior that I couldn't figure out a way around and, while it didn't break anything, I'm not exactly comfortable trying to push it through. Basically, each time you navigate to the app, the `getUser` query throws an error, the app hangs for a second and then sends you to the login page. If you remove the `{ error }` provided by Apollo, it will flash a blank MySpace (no user data at all) before redirecting to login. I tested all of this on dev and it worked, but just makes me uneasy. There are a few other things that I tried, but rather than sinking more time into it, I think we can keep the working solution we already have. All that being said, I added back in each `!user` check and its corresponding `<Loader />` and tests for redirecting an unauthorized user. Better to have a redundant layer of loading screens than risk breaking the app, or causing a bad user experience. Happy to discuss more if you'd like!


## Setup
In CMS - `yarn services:up && yarn dev`
In portal - `yarn dev`


---

## Code review steps

### As the original developer, I have

  - ... 
  - [ ] Including cypress-axe checks when UI changes
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
- For any new migrations/schema changes:

### As code reviewer(s), I have

- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):

### As a designer reviewer, I have

- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

---

## Screenshots
